Submission

AndyLinGitHub opened the pull request:
Submission

AndyLinGitHub committed:
Submission

Jackal08 commented on YuAn_Lin/selection.py:
```diff
@@ -0,0 +1,359 @@
+        self.pairwise_spearman_corr = self.ranked_daily_return.corr()
+
+    def add_target_stock(self, target_stock):
+        """Before calculating the partners of the target stock, you need to add it to this object, and determine it's potential partner combinations.
```
Would have been better if these docstrings match our format.

Jackal08 commented on YuAn_Lin/selection.py:
```diff
@@ -0,0 +1,359 @@
+from bokeh.palettes import Spectral10
+
+class PartnerSelection:
+    """Implementation of 4 partner selection methods described in Stübinger et al. (2016) section 3.1.1.
```
Good use of docstrings.

I would have made this more verbose and added a link to the original paper.

Jackal08 commented on YuAn_Lin/selection.py:
```diff
@@ -0,0 +1,359 @@
+    Parameters
+    ----------
+    daily_return : Pandas DataFrame
+        DataFrame contains adjusted close with tickers as column names and datetime as indexes
```
Grammar, you are missing full stops `.`

Jackal08 commented on YuAn_Lin/selection.py:
```diff
@@ -0,0 +1,359 @@
+        None
+        """
+
+        if target_stock not in self.ticker_list:
```
Nice, good to see this assertion.

Jackal08 commented on YuAn_Lin/selection.py:
```diff
@@ -0,0 +1,359 @@
+        if target_stock not in self.ticker_list:
+            raise KeyError("Target stock dose not exist in the object's data.")
+
+        # The paper didn't clearly describe how to determine the top N most highly correlated stocks.
```
Good use of inline comments.

Jackal08 commented on YuAn_Lin/selection.py:
```diff
@@ -0,0 +1,359 @@
+    daily_return : Pandas DataFrame
+        DataFrame contains adjusted close with tickers as column names and datetime as indexes
+    num_of_partner : int
+        Number of partner the target stock needed. if num_of_partner = n, total stocks as a cohort will be n + 1
```
This isn't clear to me?

Jackal08 commented on YuAn_Lin/selection.py:
```diff
@@ -0,0 +1,359 @@
+        self.potential_partner_dict[target_stock] = potential_partner_stocks
+        self.potential_partner_combinations_dict[target_stock] = potential_partner_combinations
+
+    def target_stock_added(self):
```
I see that all of your methods are public. Should they all be public?

Jackal08 replied:
This is a good example of a method that should be private.

Jackal08 commented on YuAn_Lin/selection.py:
```diff
@@ -0,0 +1,359 @@
+        List of selected stocks' tickers. The last element of the list will be the target stock's ticker.
+        """
+
+        if target_stock not in self.potential_partner_combinations_dict:
```
Your code needs to be more modular. You have duplicate code.

Jackal08 commented on YuAn_Lin/PartnerSelection.ipynb:
```diff
@@ -0,0 +1,964 @@
+{
+ "cells": [
```
I like how you explained your implementation step by step, showing how the user should use your implementation. 

Good use of latex. 

I would have liked it to be a bit more descriptive but overall good.

Jackal08 reviewed (approved):
Hi Yu-An Lin,

Overall good. I would have liked you to refine the quality a bit more. Will set up an interview.

* Nice write up but more could have been done to improve the quality of the document. 
* Good use of docstrings and inline comments.
* Code could be more modular. 
* All public methods. No hidden methods.
* Too much duplicate code.
* Good notebook. 
* Would have liked to see some unit tests being added. This would highlight your speed improvements.
* Would have liked some better visualizations regarding the techniques but Bokeh is a nice touch.

AndyLinGitHub committed:
Refine quality

1. Add Type Hint.
2. Add paper link.
3. Modify docstring.
4. Remove duplicate code.
4. Visualization: Add heatmap.
5. Unit test: Still working.

AndyLinGitHub commented:
I have refined some code of the PR. The main modification are listed below. I'm still working on the unit test, but it will be uploaded soon.
1. Add Type Hints.
2. Add private methods.
3. Add paper links.
4. Modify docstrings.
5. Remove duplicate code.
6. Visualization: Add heatmap.
